src/Features/LanguageServer/ProtocolUnitTests/OnAutoInsert/OnAutoInsertTests.cs

```diff
@@ -179,16 +177,79 @@ string M(int foo, bool bar)
             await VerifyMarkupAndExpected("\n", markup, expected);
         }
 
+        [Fact]
+        public async Task OnAutoInsert_BraceFormatting()
+        {
+            // The test starts with the closing brace already on a new line.
+            // In LSP, hitting enter will first trigger a didChange event for the new line character
+            // (bringing the server text to the form below) and then trigger OnAutoInsert
+            // for the new line character.
+            var markup =
+@"class A
+{
+    void M() {{|type:|}
+    }
+}";
+            var expected =
+@"class A
+{
+    void M()
+    {
+        $0
+    }
+}";
+            await VerifyMarkupAndExpected("\n", markup, expected);
+        }
+
+        [Fact]
+        public async Task OnAutoInsert_BraceFormattingInsideMethod()
+        {
+            var markup =
+@"class A
+{
+    void M()
+    {
+        if (true) {{|type:|}
+        }
+    }
+}";
+            var expected =
+@"class A
+{
+    void M()
+    {
+        if (true)
+        {
+            $0
```

Comment:
this is strange.  what if there is a C#/Vb string with this text in it?

Reply:
LSP spec on snippets - https://microsoft.github.io/language-server-protocol/specification#snippet_syntax

If there is a snippet containing both a tab stop($0) at some location and the text '$0' in a string somewhere I think we're basically hosed...

For non-snippets we're generally fine as we specify the plain text format and the client won't try to interpret the $0 as caret locations.